### PR TITLE
Adding Change Log

### PR DESCRIPTION
**Pull request by vsoch**
**Description of the Pull Request (PR):**

This PR is an effort to start being better at keeping track of changes, deprecation, and all of these import usage details that are essential for users. 

- **addition of changelog** I have added a changelog, and it dates back to the original 1.x (I did this manually and unfortunately can only grab large details). 
 - **current PRs** I have also added some soon to be merged PRs in development as an example of how we should use it. 
 - **Updated Contributing**: I separated out the sections about contributions (the PR process vs Documentation for it). I made it more clear that documentation should be added along with a note to this log.

I fixed some other small bugs, but the important thing for discussion here is the changelog. I still have no idea why the extra files for @truatpasteurdotfr are here, I can only guess it was some Github error that now remains in development. As with the other PR, please let me know to remove this or not.

**This fixes or addresses the following GitHub issues:**

- Ref: #1051 #1107

Attn: @singularityware-admin

### CODE REVIEW

File: .github/RELEASE_CHECKLIST.md
Comment (GodloveD) [inline .github/RELEASE_CHECKLIST.md:6]: Is this supposed to be here?  Does it conflict with https://github.com/singularityware/singularity/pull/1112?

File: .github/SUPPORT.md
Comment (GodloveD) [inline .github/SUPPORT.md:14]: Should really be Singularity Hub right?  There may be many Singularity Registries but there is only shub.

File: CHANGELOG.md
Comment (GodloveD) [inline CHANGELOG.md]: Seems sort of weird to shout out olifre here.

File: CHANGELOG.md
Comment (GodloveD) [inline CHANGELOG.md]: Should probably be 2.4.1 not 3.0.

File: CHANGELOG.md
Comment (GodloveD) [inline CHANGELOG.md]: Let's keep changes out until they are actually merged.  Otherwise we get into a situation where the changelog represents hopes instead of reality.

File: CHANGELOG.md
Comment (GodloveD) [inline CHANGELOG.md]: was deprecated in favor of image.expand and no longer works on images with headers (i.e. after they are built).

File: CHANGELOG.md
Comment (GodloveD) [inline CHANGELOG.md]: I would prefer if we used 3 hashes instead of bold to delineate subheadings in keeping with markdown syntax.  And as an added bonus things will work better in the future if we ever decide to add a toc and put it up online or something.

**Review by GodloveD — COMMENTED** [review]
@vsoch Thank you for putting this together!  A lot of good work in a short period of time.  For the most part this looks great.  I have some small nitpicky comments, but overall it's just what we need.

File: .github/RELEASE_CHECKLIST.md
Comment (vsoch) [reply .github/RELEASE_CHECKLIST.md:6]: They are different things - that PR adds it to the debian install files, and this one is general configuration of Singularity.

File: .github/SUPPORT.md
Comment (vsoch) [reply .github/SUPPORT.md:14]: No, lol. Singularity Registry is a different thing :)

https://github.com/singularityhub/sregistry

it's even peer reviewed / published!

http://joss.theoj.org/papers/050362b7e7691d2a5d0ebed8251bc01e

it's times like these the stuff I do feels... totally unnoticed lol.

File: CHANGELOG.md
Comment (vsoch) [reply CHANGELOG.md]: why? It's his statements. If I don't it's plagarism.

File: CHANGELOG.md
Comment (vsoch) [reply CHANGELOG.md]: @gmkurtzer said the next "official" version is 3.0, but I suppose we didn't plan for the bug fixes. I'll change it if it makes you happy.

File: CHANGELOG.md
Comment (vsoch) [reply CHANGELOG.md]: aye aye cap-i-tan!

File: CHANGELOG.md
Comment (vsoch) [reply CHANGELOG.md]: done

File: .github/RELEASE_CHECKLIST.md
Comment (GodloveD) [reply .github/RELEASE_CHECKLIST.md:6]: OK thanks.

File: .github/SUPPORT.md
Comment (GodloveD) [reply .github/SUPPORT.md:14]: Oh sorry.  I thought shub was a special case of a Singularity registry.  Nevermind.

File: CHANGELOG.md
Comment (GodloveD) [reply CHANGELOG.md]: Because this is basically documentation and it seems unprofessional.  But if you feel that you would be a plagiarist without the proper citation I won't argue.  I don't feel strongly about it.

File: CHANGELOG.md
Comment (GodloveD) [reply CHANGELOG.md]: It's either that or go without a changelog in 2.4.1.   😺

File: .github/SUPPORT.md
Comment (vsoch) [reply .github/SUPPORT.md:14]: It's like homo sapiens and Neanderthal. Maybe some wayward, back in the day common elements (mostly UI) but different beasties under it all!

Comment (vsoch) [reply .github/SUPPORT.md:14]: but yeah Shub (which came first) does deliver the common API endpoints of a registry so Shub works like one too. It's the mamma registry! The HUB.

File: CHANGELOG.md
Comment (GodloveD) [reply CHANGELOG.md]: https://www.youtube.com/watch?v=46goXXCm9Fo

File: CHANGELOG.md
Comment (vsoch) [reply CHANGELOG.md]: I removed it! 5 hours ago :)

**Review by GodloveD — APPROVED** [review]
This looks good to me.  @gmkurtzer said we can go ahead and merge this when ready.  So I think it's ready.  Can you squash the commits or whatever and merge?

**Comment by vsoch** [comment]
Sure thing! I'll wait for it to turn green. My trust in Travis consistency is... squishy at best.

**Comment by vsoch** [comment]
Travis is being wanky, so since there isn't actually any code to test, I'm going to squash and merge. Onwards to a new beginning with better documentation, long live CHANGELOG.md!

**Pull request merged**
